# Kinase LID Redesign

**BioPipelines example** — de novo backbone design of the adenylate kinase LID domain using RFdiffusion, followed by ProteinMPNN sequence design and AlphaFold2 refolding. Candidates are filtered by RMSD of the fixed regions and pLDDT.

[![Documentation](https://img.shields.io/badge/docs-readthedocs-blue)](https://biopipelines.readthedocs.io/en/latest/)
[![Preprint](https://img.shields.io/badge/preprint-bioRxiv-B31B1B)](https://www.biorxiv.org/content/10.64898/2026.03.11.711024v1)

In [1]:
# Cell 1: Install BioPipelines and micromamba
# !git clone https://github.com/locbp-uzh/biopipelines
# %cd biopipelines
from getpass import getpass
tok_name = getpass("Token name: ")
tok = getpass("Token value: ")
!git clone -b main https://{tok_name}:{tok}@gitlab.uzh.ch/locbp/public/biopipelines-locbp.git
%cd biopipelines-locbp
!pip install -e ".[colab]"
!wget -q https://github.com/mamba-org/micromamba-releases/releases/latest/download/micromamba-linux-64 -O /usr/local/bin/micromamba && chmod +x /usr/local/bin/micromamba

Token name: ··········
Token value: ··········
Cloning into 'biopipelines-locbp'...
remote: Enumerating objects: 8438, done.
remote: Counting objects: 100% (47/47), done.
remote: Compressing objects: 100% (47/47), done.
remote: Total 8438 (delta 20), reused 0 (delta 0), pack-reused 8391 (from 1)
Receiving objects: 100% (8438/8438), 10.80 MiB | 6.08 MiB/s, done.
Resolving deltas: 100% (6422/6422), done.
/content/biopipelines-locbp
Obtaining file:///content/biopipelines-locbp
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.2/3.2 MB 68.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.1/37.1 MB 76.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.1/118.1 kB 15.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.1/16.1 MB 93.7 MB/s eta 0:00:

In [2]:
# Cell 2: Mount Google Drive and repoint BioPipelines folders
from google.colab import drive
drive.mount('/content/drive')
!bp-config set folders.base.biopipelines_output /content/drive/MyDrive/BioPipelines
!bp-config set folders.base.data /content/drive/MyDrive/BioPipelines/data
!bp-config set folders.infrastructure.cache /content/drive/MyDrive/BioPipelines/cache

Mounted at /content/drive
folders.base.biopipelines_output: '/content/BioPipelines' -> '/content/drive/MyDrive/BioPipelines'  (/content/biopipelines-locbp/config.colab.yaml, backup: config.colab.yaml.bak)
folders.base.data: '/content/data' -> '/content/drive/MyDrive/BioPipelines/data'  (/content/biopipelines-locbp/config.colab.yaml, backup: config.colab.yaml.bak)


In [3]:
# Cell 2: Install tools
from biopipelines.pipeline import *
from biopipelines import RFdiffusion, ProteinMPNN, AlphaFold, PyMOL

with Pipeline(project="Setup", job="InstallTools"):
    RFdiffusion.install()
    ProteinMPNN.install()
    AlphaFold.install()
    PyMOL.install()


Running RFdiffusion_installation (step 1)
=== Installing RFdiffusion ===
File ‘Base_ckpt.pt’ already there; not retrieving.

File ‘Complex_base_ckpt.pt’ already there; not retrieving.

Fetch Shard Index for conda-forge/linux-64                                                      ⧖ Starting
Fetch Shard Index for conda-forge/linux-64                                                ✔ Done (0.1 sec)
Fetch Shard Index for conda-forge/noarch                                                        ⧖ Starting
Fetch Shard Index for conda-forge/noarch                                                  ✔ Done (0.1 sec)
Fetching and Parsing Packages' Shards                                                           ⧖ Starting
Fetching and Parsing Packages' Shards                                                     ✔ Done (2.0 sec)
Using Cached Shard Index for conda-forge/linux-64                                                   ✔ Done
Using Cached Shard Index for conda-forge/noarch                  

## Cell 3: Kinase LID Redesign Pipeline

Takes adenylate kinase (PDB: 4AKE) and redesigns the LID domain (residues 118–160, replaced with a new 50–70 residue loop).
The top 3 designs with RMSD < 1.5 Å on the fixed scaffold and highest pLDDT are selected.

In [4]:
# Cell 3: Pipeline
from biopipelines.pipeline import *
from biopipelines import RFdiffusion,ProteinMPNN, AlphaFold, ConformationalChange, Panda

with Pipeline(project="AdenylateKinase", job="LID_Redesign"):
    kinase = PDB("4AKE")
    backbones = RFdiffusion(pdb=kinase,
                            contigs='A1-117/50-70/A161-214',
                            num_designs=3)
    sequences = ProteinMPNN(structures=backbones,
                            num_sequences=2,
                            redesigned=backbones.tables.structures.designed)
    refolded = AlphaFold(proteins=sequences)
    conf_change = ConformationalChange(reference_structures=kinase,
                                       target_structures=refolded,
                                       selection=backbones.tables.structures.fixed)
    top3 = Panda(tables=[refolded.tables.confidence,
                          conf_change.tables.changes],
                 operations=[Panda.merge(),
                              Panda.filter("RMSD < 1.5"),
                              Panda.sort("plddt", ascending=False),
                              Panda.head(3)],
                 pool=refolded)
top3

  4AKE not found locally, will download from RCSB

Running PDB (step 1)
Fetching 1 structures
Convert: keep as-is (pdb|cif)
PDB IDs: 4AKE
Custom IDs: 4AKE
Priority: pdbs/ -> RCSB download
Output folder: /content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/001_PDB
Fetching 1 structures (keep as-is (pdb|cif))
Priority: pdbs/ -> RCSB download
Water molecules will be removed

[1/1] Processing 4AKE -> 4AKE
4AKE not found locally, downloading from RCSB
Cached to pdbs/ folder: /content/biopipelines-locbp/pdbs/4AKE.pdb
Successfully downloaded 4AKE as 4AKE: 297432 bytes (rcsb_download (PDB))

Successful fetches saved: /content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/001_PDB/structures/structures.csv (1 structures)
Sequences saved: /content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/001_PDB/sequences/sequences.csv (1 sequences)
No ligands found - created empty table: /content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/001_PDB/

StandardizedOutput({'structures': DataStream(name='structures', format='pdb', items=3, files=3, map_table=set), 'msas': DataStream(name='msas', format='a3m', items=3, files=3, map_table=set), 'rendering_parameters': {'structures': {'color_by': 'plddt', 'plddt_upper': 100}}, 'tables': {'result': TableInfo(name='result', path='/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/006_Panda/tables/merge_filter_sort.csv', columns=['id', 'structure', 'plddt', 'max_pae', 'ptm']), 'missing': TableInfo(name='missing', path='/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/006_Panda/tables/missing.csv', columns=['id', 'removed_by', 'cause']), 'structures': TableInfo(name='structures', path='/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/006_Panda/structures/structures_map.csv', columns=['id', 'file']), 'confidence': TableInfo(name='confidence', path='/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/006_Panda/tables/confidence.csv', columns=['id', 'structure', 'plddt', 'max_pae', 'ptm']), 'msas': TableInfo(name='msas', path='/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/006_Panda/msas/msas_map.csv', columns=['id', 'sequences.id', 'sequence', 'msa_file'])}, 'output_folder': '/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/006_Panda'})

In [5]:
backbones

StandardizedOutput({'structures': DataStream(name='structures', format='pdb', items=3, files=1, map_table=set), 'tables': {'structures': TableInfo(name='structures', path='/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/002_RFdiffusion/tables/structures.csv', columns=['id', 'pdb', 'fixed', 'designed', 'source_fixed', 'plddt_mean', 'status'])}, 'output_folder': '/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/002_RFdiffusion'})

In [6]:
sequences

StandardizedOutput({'sequences': DataStream(name='sequences', format='csv', items=6, files=0, map_table=set), 'fasta': DataStream(name='fasta', format='fasta', items=6, files=1(shared), map_table=set), 'tables': {'sequences': TableInfo(name='sequences', path='/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/003_ProteinMPNN/sequences/sequences.csv', columns=['id', 'structures.id', 'source_pdb', 'sequence', 'score', 'seq_recovery', 'gaps']), 'missing': TableInfo(name='missing', path='/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/003_ProteinMPNN/tables/missing.csv', columns=['id', 'removed_by', 'cause'])}, 'output_folder': '/content/drive/MyDrive/BioPipelines/AdenylateKinase/LID_Redesign_001/003_ProteinMPNN'})